lib/client-options.js

```diff
@@ -44,7 +44,8 @@ function defaultOptions () {
     authProvider: null,
     maxPrepared: 500,
     encoding: {
-      copyBuffer: true
+      copyBuffer: true,
+      useUndefinedAsUnset: true
```

Comment:
You are right, `useUndefinedAsUnset` affected protocol versions < 4.
The default behaviour in driver v2.1 (before that setting was introduced) was that `undefined` was encoded as `null`. I've restored the same behaviour by [checking protocol version](https://github.com/datastax/nodejs-driver/pull/103/files#diff-b364beb377c8302c49a6988ecb79bc26R170).